assets/tests/components/propertiesPanel/blockWaivers.test.tsx

```diff
@@ -0,0 +1,152 @@
+import React from "react"
+import renderer from "react-test-renderer"
+import BlockWaivers, * as blockWaivers from "../../../src/components/propertiesPanel/blockWaivers"
+import { BlockWaiver } from "../../../src/realtime"
+
+const {
+  CurrentFuturePastType,
+  currentFuturePastClass,
+  currentFuturePastTitle,
+  currentFuturePastType,
+  formatTimeOfDay,
+  hours12,
+  nowTimeOfDay,
+} = blockWaivers
+
+describe("BlockWaivers", () => {
```

Comment:
I think snapshots for a past waiver and a future waiver would test the same functionality as some of the whitebox tests below without needing to expose and be sensitive to so much implementation detail.

Reply:
I'm trying to make sure I'm understanding this properly. Are you referring to the tests of various functions, e.g. `formatTimeOfDay`? Those were very useful to me in developing those functions and I think would still serve an important purpose if we wanted to refactor. But I think more importantly, are you suggesting there are more rendering tests you'd like to see?

Reply:
The time formatting makes sense to have unit tests for, since there's a lot of detail there that you don't want to have to cover in the bigger `BlockWaiverBanner` tests.

I'm thinking about `currentFuturePastType()`, `currentFuturePastClass()`, and `currentFuturePastTitle()` tests. In all of those functions, all 3 branches would be covered by `BlockWaiverBanner` tests for the past, present, and future.

Reply:
Ah, got it. Added rendering tests for all three cases.